doc: Add docs for `EthApi`

tcoratger opened the pull request:
Should close #9540.

mattsse reviewed (approved):
lgtm, pending @emhane

emhane commented:
this is too much chagpt fluff

tcoratger commented:
Hey @emhane,

Just to tell you that in fact I used ChatGPT (that seems normal to me for generating documentation in 2024, right?). However, I spent about 30 minutes writing the entire `OpEthApi` example (in documentation form) with also the definition for myself of each helper trait:

```rust
/// - `EthApiSpec`: Defines basic Ethereum network information and status.
/// - `SpawnBlocking`: Handles task spawning and concurrency.
/// - `LoadFee`: Manages fee-related data loading and caching.
/// - `Call`: Handles EVM call operations.
/// - `LoadState`: Manages state loading and caching.
/// - `EthState`: Provides Ethereum state-related functionality.
/// - `EthCall`: Implements Ethereum call-related methods.
/// - `EthFees`: Handles fee estimation and history.
/// - `Trace`: Provides tracing functionality for transactions.
/// - `AddDevSigners`: Manages development signers for testing.
/// - `UpdateRawTxForwarder`: Allows updating of raw transaction forwarding behavior.
```

that ChatGPT cannot guess (at least not from what I understood). Then in fact when I pass all this context to ChatGPT (i.e. more or less all the main documentation of the `EthApi`), he is much better than me at organizing the pieces and writing clear documentation (and I imagine that no matter who by the way).

But in general, I don't really understand why you close PRs without review and feedback loops. Every time I feel like I'm doing a lot of work for nothing. This is not a criticism at all, if you feel that this PR should not be merged, you are the one responsible for the code, I can only agree but I would like to understand better:
- On this PR we could have removed the parts that don't suit you, shortened it, removed information, added what seemed missing to you.
- Here https://github.com/paradigmxyz/reth/pull/8861 it's the same I didn't really understand because there was no ChatGPT and a lot of super useful work to do on it. I saw a comment saying that we probably had to modify the `BundleBuilder` API but that's not a problem, we could have done it. But ultimately it was abandoned and everything was closed?

cc @mattsse
